tests/Doctrine/Migrations/Tests/Generator/DiffGeneratorTest.php

```diff
@@ -92,15 +96,42 @@ static function ($name): bool {
             ->method('dropTable')
             ->will(self::onConsecutiveCalls('schema.table_name2', 'schema.table_name3'));
 
-        $fromSchema->expects(self::once())
-            ->method('getMigrateToSql')
-            ->with($toSchema, $this->platform)
-            ->willReturn(['UPDATE table SET value = 2']);
+        if (method_exists($this->schemaManager, 'createComparator')) {
+            $schemaDiff = $this->createStub(SchemaDiff::class);
+            $schemaDiff->method('toSql')->willReturn($this->onConsecutiveCalls(
+                ['UPDATE table SET value = 2'],
+                ['UPDATE table SET value = 1']
+            ));
+
+            // regular mocks cannot be used here, because the method is static
+            $comparator = new class extends Comparator {
+                /** @var SchemaDiff */
+                public static $schemaDiff;
+
+                public static function compareSchemas(
+                    Schema $fromSchema,
+                    Schema $toSchema
+                ): SchemaDiff {
+                    return self::$schemaDiff;
+                }
+            };
+
+            $comparator::$schemaDiff = $schemaDiff;
 
-        $fromSchema->expects(self::once())
-            ->method('getMigrateFromSql')
-            ->with($toSchema, $this->platform)
-            ->willReturn(['UPDATE table SET value = 1']);
+            $this->schemaManager->expects(self::once())
+                ->method('createComparator')
+                ->willReturn($comparator);
+        } else {
+            $fromSchema->expects(self::once())
+                ->method('getMigrateToSql')
+                ->with($toSchema, $this->platform)
+                ->willReturn(['UPDATE table SET value = 2']);
+
+            $fromSchema->expects(self::once())
+                ->method('getMigrateFromSql')
+                ->with($toSchema, $this->platform)
+                ->willReturn(['UPDATE table SET value = 1']);
+        }
```

Comment:
I tried to find a way to abide by "don't mock what you don't own" and reduce the big amount of mocking in this test, and gave up… I think it should be reworked, but I'm not sure how.